core/types/receipt.go

```diff
@@ -438,5 +458,26 @@ func (rs Receipts) DeriveFields(config *params.ChainConfig, hash common.Hash, nu
 			logIndex++
 		}
 	}
+	if config.Optimism != nil && len(txs) >= 2 { // need at least an info tx and a non-info tx
+		if data := txs[0].Data(); len(data) >= 4+32*8 { // function selector + 8 arguments to setL1BlockValues
```

Comment:
This doesn't handle versioning but thats ok for now

Reply:
Thanks for checking the update path. I think we can use the function selector bytes, or read unused bytes, or use the block number, or update in other different ways. Enough options for future updates